### PR TITLE
Moved libstatistics_collector folder from system_metrics_collector

### PR DESCRIPTION
**Pull request by dabonnie**
Related to: https://github.com/ros-tooling/aws-roadmap/issues/248

Note: this is just a move and CI has not been added yet, because CI depends on https://github.com/ros-tooling/metrics_statistics_msgs/pull/1. CI will be added in another PR.

Signed-off-by: Devin Bonnie <dbbonnie@amazon.com>

### CODE REVIEW

**Comment by thomas-moulard** [comment]
+1 let's merge the other PR first

**Comment by dabonnie** [comment]
SGTM, I'll mark the dependencies clearly on the tickets.

File: README.md [hunk 1]
```diff
@@ -1,3 +1,9 @@
 # libstatistics_collector
 
-ROS 2 package providing utility classes to calculate and publish statistics about collected metric measurements.
+
+## Description
+
+This C++ library provides the following:
+- A `Collector` interface for implementing classes that collect observed data and generate statistics for them
+- A `TopicStatisticsCollector` interface for implmenenting classes that collect and perform measurements for ROS2 topic statistics
```
Comment (prajakta-gokhale) [inline README.md:8]: Should we also mention the message age and period implementations?

File: include/libstatistics_collector/collector/collector.hpp [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+
+#ifndef LIBSTATISTICS_COLLECTOR__COLLECTOR__COLLECTOR_HPP_
+#define LIBSTATISTICS_COLLECTOR__COLLECTOR__COLLECTOR_HPP_
+
+#include <mutex>
+#include <string>
+
+#include "metric_details_interface.hpp"
+#include "libstatistics_collector/moving_average_statistics/moving_average.hpp"
+#include "libstatistics_collector/moving_average_statistics/types.hpp"
```
Comment (prajakta-gokhale) [inline include/libstatistics_collector/collector/collector.hpp:23]: Alphasort

**Comment by prajakta-gokhale** [comment]
`ament_copyright` is failing because it cannot find the LICENSE (which is added in this PR).

**Pull request merged**